src/spu/src/services/public/stream_fetch.rs

```diff
@@ -312,103 +302,100 @@ impl StreamFetchHandler {
+                    // we mark last offset in the response that we should sync up
+                    ..Default::default()
                 };
 
-                Ok((next_offset, consumer_wait))
-            } else {
-                debug!("no filter, sending back entire");
-
-                let response = StreamFetchResponse {
+                let filter_response = StreamFetchResponse {
                     topic: self.replica.topic.clone(),
                     stream_id: self.stream_id,
-                    partition: file_partition_response,
+                    partition: filter_partition_response,
                 };
 
-                let response_msg = RequestMessage::<FileStreamFetchRequest>::response_with_header(
-                    &self.header,
-                    response,
-                );
+                let filter_response_msg =
+                    RequestMessage::<DefaultStreamFetchRequest>::response_with_header(
+                        &self.header,
+                        filter_response,
+                    );
 
-                trace!("sending back file fetch response msg: {:#?}", response_msg);
+                trace!("sending back filter respone: {:#?}", filter_response_msg);
 
                 let mut inner_sink = self.sink.lock().await;
                 inner_sink
-                    .encode_file_slices(&response_msg, self.header.api_version())
+                    .send_response(&filter_response_msg, self.header.api_version())
                     .await?;
+                true
+            } else {
+                debug!(next_offset, "filter, no records send back, skipping");
+                false
+            };
 
-                drop(inner_sink);
+            Ok((next_offset, consumer_wait))
+        } else {
+            debug!("no filter, sending back entire");
 
-                debug!(read_time_ms = %now.elapsed().as_millis(),"finish sending back records");
+            let response = StreamFetchResponse {
+                topic: self.replica.topic.clone(),
+                stream_id: self.stream_id,
+                partition: file_partition_response,
+            };
 
-                Ok((offset.isolation(&self.isolation), true))
-            }
-        } else {
-            debug!("empty records, skipping");
+            let response_msg = RequestMessage::<FileStreamFetchRequest>::response_with_header(
+                &self.header,
+                response,
+            );
+
+            trace!("sending back file fetch response msg: {:#?}", response_msg);
+
+            let mut inner_sink = self.sink.lock().await;
+            inner_sink
+                .encode_file_slices(&response_msg, self.header.api_version())
+                .await?;
+
+            drop(inner_sink);
+
+            debug!(read_time_ms = %now.elapsed().as_millis(),"finish sending back records");
 
-            Ok((offset.isolation(&self.isolation), false))
+            Ok((offset.isolation(&self.isolation), true))
```

Comment:
This diff is misleading, I didn't just change a true to false. This `true` is actually from this line in the original source:

https://github.com/infinyon/fluvio/pull/1188/files#diff-5bcc5ec4673d6a5c8633195a7360c3402f873769c1faa90cf627b242179d2241L406